### DISCUSSION TITLE
E-mail Address Masking

### DISCUSSION THREAD

**Discussion by windhandel** [beholdr/maska#173] [Q&A]
I've reviewed all of the online examples I can put my hands on. Search SO specifically for results and tested a variety of ways to try and do an email address mask.

It's the @ symbol that I'm having a hard time with. Here's what I've tried:

`v-maska data-maska="ZAZ.Z" data-maska-tokens="Z:\w:multiple|A:[@@]" `

The double @@ is because this is within an ASP.NET Razor View.

Any help would be appreciated.

Thanks,
Jeff

**Comment by Vasek18** [comment]
+1
I can't get how to make an email mask

**Reply by Vasek18** [reply]
So far the best idea I've come up with

mask: Yyyyyyyyyyyyyyyyyyyyyyyyy!@Yyyyyyyyyyyyyyyyyyyyyyyyy.YYy
tokens: y:[A-Za-z0-9а-яА-Я\-\_]:optional|Y:[A-Za-z0-9а-яА-Я\-\_]

But it's sick

**Comment by zourtney** [comment]
Ugh. Ok. I'm in the same situation.

I can't believe I'm doing this... but @Vasek18, I've modified your self-proclaimed "sick" solution, allowing for a few other legal addresses and capping at the (possible) [maximum email address length](https://www.rfc-editor.org/errata/eid1690#:~:text=Section%203%20says%3A,total%20length%20of%20320%20characters) of 64 characters of "local" and 256 characters of "domain".

Avert your eyes. I need a better solution.

```
export const emailMask: MaskOptions = {
  mask: 'LlllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllADdddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd',
  tokens: {
    L: { pattern: /[A-Za-z0-9а-яА-Я-_.]/ },
    l: { pattern: /[A-Za-z0-9а-яА-Я-_.]/, optional: true },
    A: { pattern: /[@]/ },
    D: { pattern: /[0-9a-zA-Z\-.]/ },
    d: { pattern: /[0-9a-zA-Z\-.]/, optional: true },
  },
  eager: true,
};

```

NOTE: I can never get this to register as "completed" unless _every_ optional character is present *and* the character `@` is registered as a separate token...yeah I don't know.

NOTE2: it doesn't catch tricky situations like doubled `.` or `-`. Meh.

**Reply by guronos** [reply]
maybe
mask: "A!@B!.C",
                eager: true,
                tokens: {
                    A: { pattern: /[a-zA-Z0-9._-]/, multiple: true },
                    B: { pattern: /[a-zA-Z0-9_-]/, multiple: true },
                    C: { pattern: /[a-zA-Z0-9_-]/, multiple: true },
                }
